src/state/session/index.tsx

```diff
@@ -196,18 +193,20 @@ export function Provider({children}: React.PropsWithChildren<{}>) {
       const syncedAccount = synced.accounts.find(
         a => a.did === synced.currentAccount?.did,
       )
+
       if (syncedAccount && syncedAccount.refreshJwt) {
         if (syncedAccount.did !== state.currentAgentState.did) {
           resumeSession(syncedAccount)
         } else {
           const agent = state.currentAgentState.agent as BskyAgent
           const prevSession = agent.session
-          agent.session = sessionAccountToSession(syncedAccount)
+          const nextSession = sessionAccountToSession(syncedAccount)
+          agent.resumeSession(nextSession)
           addSessionDebugLog({
             type: 'agent:patch',
             agent,
             prevSession,
-            nextSession: agent.session,
+            nextSession,
           })
         }
```

Comment:
Slightly modified since we can't set `agent.session` this way anymore (and don't want to rm since it looks like a lot of users are hitting this...). However, wondering about one thing...

`agent.session` can possibly be undefined - and I think likely would be here if we call `agent.resumeSession` without `await`. Would we have enough info for the log by just passing it the session from `sessionAccountToSession()`?